### PR TITLE
feature/usage-instructions

### LINKED ISSUE #1387: Usage scenarios or usage instructions to Api details page
Include simple usage instructions or common usage instructions to API details page. These instructions should contain a possibility to copy at least user's API key (if an API key is available and if a proxy has been selected to this particular API)

# User story
> As an API consumer
I want to be able to quickly find out how to use the API I'm interested in through APInf
so that I do not have to find the information from several different places

# Definition of done
* [x] Select what would be the two most common / relevant methods of making api calls
  * **url parameter** - with [`?api_key=###` query parameter](https://api-umbrella.readthedocs.io/en/latest/getting-started.html#make-an-api-request)
  * **request header** - passing in [`X-Api-Key` as a request header](https://api-umbrella.readthedocs.io/en/latest/api-consumer/api-key-usage.html#http-header)
* [ ] Document the methods in user instructions
* [ ] Add the usage instructions to the API profile view (Details tab)
* [ ] Auto-insert e.g. base URL and user's API key.  (API key only if a proxy has been selected)
* [ ] Allow copying base URL and user's API key, if available.

# Wireframe

apidetailsinstruction1

### PR DESCRIPTION
**Pull request by nnn**
Closes #1387

### CODE REVIEW

**Comment by nnn** [comment]
Blocked by #1520

**Comment by bajiat** [comment]
@nnn What is the status?

**Comment by nnn** [comment]
@bajiat, working on this today.

**Comment by nnn** [comment]
Removed text.
Added test call button.

**Comment by brylie** [comment]
Based on in-office review, this task needs further work. Please follow the guidelines outlined in the *definition of done*.

File: apis/client/view/details/details.html [hunk 1]
```diff
@@ -40,14 +40,25 @@ <h2 class="panel-title clearfix">
...
-        <a href="{{ testCallUrl }}" class="btn btn-success test-call-button">
-          API test call
-        </a>
-        {{ / if }}
       </div>
     </div>
   </div>
 </div>
+<div id="usage-instructions" class="bs-callout bs-callout-primary">
+  <h4>To make calls to this API, you can either:</h4>
```
Comment (brylie) [inline apis/client/view/details/details.html]: Add i18n token here.

File: apis/client/view/details/details.html [hunk 2]
```diff
@@ -40,14 +40,25 @@ <h2 class="panel-title clearfix">
...
+  <ul>
+    <li>
+      call the API with your key in the URL parameters:
```
Comment (brylie) [inline apis/client/view/details/details.html]: Add i18n token here.

File: apis/client/view/details/details.html [hunk 3]
```diff
@@ -40,14 +40,25 @@ <h2 class="panel-title clearfix">
...
+      <span>
+        https://&#x0003C;host&#x0003E;/&#x0003C;base_path&#x0003E;?api_key=&#x0003C;user_api_key&#x0003E;
```
Comment (brylie) [inline apis/client/view/details/details.html]: This needs to generate a valid URL for the user by using actual values from API Umbrella for *host*, *base_path*, and *api_key*.

We have had reference code for this in the past. Basically, we are re-adding a feature that was removed about two months ago.

File: apis/client/view/details/details.html [hunk 4]
```diff
@@ -40,14 +40,25 @@ <h2 class="panel-title clearfix">
...
+      </span>
+    </li>
+    <li>
+      call the API with your key in the request headers:
```
Comment (brylie) [inline apis/client/view/details/details.html]: Add i18n token here.

File: apis/client/view/details/details.html [hunk 5]
```diff
@@ -40,14 +40,25 @@ <h2 class="panel-title clearfix">
...
+      <span>
+        X-Api-Key: &#x0003C;user_api_key&#x0003E;
```
Comment (brylie) [inline apis/client/view/details/details.html]: This needs to use the actual value for *user_api_key*

File: apis/client/view/details/details.js [hunk 6]
```diff
@@ -46,44 +46,4 @@ Template.apiDetails.helpers({
 
     return url;
   },
-  testCallUrl () {
```
Comment (brylie) [inline apis/client/view/details/details.js]: This function contains code that is helpful to generate the test call url. Please use parts of this code to construct a valid test call url.

**Comment by nnn** [comment]
@brylie updated, check what do you think?

**Comment by nnn** [comment]
@brylie please review. I think now it's ready to be merged.

**Comment by nnn** [comment]
@brylie I will ask @frenchbread to merge this since you've added some commits.

**Comment by frenchbread** [comment]
Looks sweet! Merging..

**Pull request merged**